tumblr-downloader.go

```diff
@@ -150,67 +154,110 @@ func tumblrGetLikes(httpClient *http.Client, blogIdentifier string, timestamp in
 	return parser
 }
 
-func downloadURL(URL string) {
-	resp, err := http.Get(URL)
-	checkError(err)
+func downloadURL(URL string, SubDir string, AppendExtension string) {
+	fmt.Printf("Downloading %s ... ",URL)
 
-	tokens := strings.Split(URL, "/")
-	fileName := tokens[len(tokens)-1]
+	//tokens := strings.Split(URL, "/")
+	//fileName := tokens[len(tokens)-1]
+	fileName := reURL.FindString(URL)
+	if fileName != "" {
 
-	os.Mkdir(outputFolderName, os.ModePerm)
+		outputFilePath := outputFolderName + string(os.PathSeparator) + SubDir + fileName + AppendExtension
+		os.Mkdir(outputFolderName + string(os.PathSeparator) + SubDir, os.ModePerm)
+		//fmt.Println(outputFilePath)
 
-	outputFilePath := outputFolderName + string(os.PathSeparator) + fileName
+		_, err := os.Stat(outputFilePath)
+		if os.IsNotExist(err) {
+			resp, err := http.Get(URL)
+			checkError(err)
 
-	if _, err = os.Stat(outputFilePath); os.IsNotExist(err) {
-		outFile, err := os.Create(outputFilePath)
-		checkError(err)
+			outFile, err := os.Create(outputFilePath)
+			checkError(err)
 
-		defer outFile.Close()
-		_, err = io.Copy(outFile, resp.Body)
+			defer outFile.Close()
+			_, err = io.Copy(outFile, resp.Body)
+			fmt.Println("done")
+		} else {
+			fmt.Println("File already exists, skipping")
+		}
 	} else {
-		fmt.Println("File already exists, skipping.")
+		fmt.Println("No tumblr link, skipping")
 	}
+
 }
 
 func main() {
+	fmt.Printf("Initializing environment... ")
 	loadConfig()
 	initOauthConfig()
+	reVideo, err := regexp.Compile("src=\"([^\"]+)\"")
+	checkError(err)
+	fmt.Println("done")
 
+	fmt.Printf("Authorizing on tumblr... ")
 	httpClient := authTumblr()
 	lastTimestamp := 0
+	fmt.Println("success")
 
 	counter := 1
 	for {
+		fmt.Printf("Ask for liked posts %d - %d... ",counter,counter+limit-1)
```

Comment:
Really small nitpicking, but can we put space after arguments, or even better run `go fmt` on the whole file?